extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/CurrentRequestProducer.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.vertx.http.runtime;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Note that we intentionally put the producer in a separate class so that it's possible to exclude the bean if a naming
+ * conflict exists.
+ */
+@Singleton
+public class CurrentRequestProducer {
+
+    @Named("request")
```

Comment:
Are we sure we want it to be named `request` instead of `vertx-request`?

Reply:
I'm not sure at all ;-). `vertx-request` or `vertxRequest` would be more accurate and less likely to collide but it's also longer to type. But I think you're right. Let's change the name to `vertxRequest`...